dashboard/src/components/AppList/AppList.test.tsx

```diff
@@ -57,15 +56,25 @@ context("when changing props", () => {
     jest.spyOn(qs, "parse").mockReturnValue({
       q: "foo",
     });
-    const wrapper = mountWrapper(defaultStore, <AppList />);
+    const wrapper = mountWrapper(
+      defaultStore,
+      <MemoryRouter initialEntries={["/foo?q=foo"]}>
+        <AppList />
+      </MemoryRouter>,
```

Comment:
The main change here is replacing the jest.spyOn with the approach suggested in the official website: https://reactrouter.com/web/guides/testing